### ISSUE TITLE
https://numpy.org/doc/stable/index.html shows version 1.21.0.dev0+279.g5d54729ddc

### ISSUE THREAD

**Issue by mattip** [numpy/numpy#18265]
Something is a bit off with the 1.20 documentation. The front page https://numpy.org/doc/stable/index.html is showing a non-release version number.

**Comment by rgommers** [comment]
This looks like a mistake during doc build uploading for 1.20.0. @charris did you do that upload?

**Comment by charris** [comment]
> did you do that upload?

Yes. The problem was that I was unable to build the docs before updating the release branch for further development. I thought that might be a problem but didn't see it after a quick check. I think the fix is to just check out the proper commit before the build and that was already on my list for today, the problem wasn't unexpected :)

**Comment by charris** [comment]
The problem is that the documentation was built against my installed NumPy. I thought it was supposed to be built in its own environment. I can fix that, but it either needs fixing or to be in the notes.

**Comment by rgommers** [comment]
> EDIT: That is one reason I'd like to see documentation builds made part of runtests, it already sets up an appropriate environment.

I'd like to build them in CI together with wheels, much more controlled. And same for the sdist, for SciPy we recently had an issue with that (https://github.com/scipy/scipy/issues/13369), that could easily happen for NumPy as well.

**Comment by rgommers** [comment]
> The problem is that the documentation was built against my installed NumPy. I thought it was supposed to be built in its own environment. I can fix that, but it either needs fixing or to be in the notes.

`make dist` is indeed supposed to do that. Maybe you had `PYTHONPATH` defined that was overriding it or something.

**Comment by charris** [comment]
> `make dist` is indeed supposed to do that. Maybe you had `PYTHONPATH` defined

IIRC, `make dist` has never worked that way for me. I don't have `PYTHONPATH` in my environment. OTOH, NumPy is installed in `~/.local/lib/python3.9/site-packages`, which might be a problem for some reason.

**Comment by charris** [comment]
Having the `.local` install wasn't a problem, the newly compiled numpy version was properly installed, but could not be found. So this seems like a path problem. I just installed the 1.20.0 release in `.local`  and everything was OK except for an extra compile step.

**Comment by melissawm** [comment]
Looks like this has been fixed - I'll close but feel free to reopen if that's not right.
